### PR TITLE
[feature](executor)  Automatically set the instance_num using the info from be.

### PR DESCRIPTION
**Pull request by Mryange**
# Proposed changes

1. fixed some error regressions (results error with big nstance_num due to incorrect order by).
2. if set parallel_fragment_exec_instance_num to 0, the concurrency in the Pipeline execution engine will automatically be set to half of the number of CPU cores.
3. add limit to parallel_fragment_exec_instance_num that it cannot be set to more than fe.conf::max_instance_num(Default: 128)
```
mysql [(none)]>set parallel_fragment_exec_instance_num = 514;
ERROR 1231 (42000): errCode = 2, detailMessage = Variable 'parallel_fragment_exec_instance_num' can't be set to the value of '514(Should not be set to more than 128)'
```
if set max_instance_num = 100 in fe.conf
image

```
mysql [(none)]>set parallel_fragment_exec_instance_num = 114;
ERROR 1231 (42000): errCode = 2, detailMessage = Variable 'parallel_fragment_exec_instance_num' can't be set to the value of '114(Should not be set to more than 100)'
```
Collect info through be's disk report (disk information updates every minute).
## Problem summary

In the past, instance_num was generally set to a random number between 1 and 8 (fuzzy). The CPU core number of the test machine was 32 and would be set to 16, which would not affect the correctness.
https://github.com/apache/doris/pull/19324
## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.46 seconds
 stream load tsv:          426 seconds loaded 74807831229 Bytes, about 167 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230507074838_clickbench_pr_139755.html

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p1

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SetVar.java [hunk 1]
```diff
@@ -155,7 +155,14 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
             this.value = new StringLiteral(TimeUtils.checkTimeZoneValidAndStandardize(getValue().getStringValue()));
             this.result = (LiteralExpr) this.value;
         }
-
+        if (getVariable().equalsIgnoreCase(SessionVariable.PARALLEL_FRAGMENT_EXEC_INSTANCE_NUM)) {
+            int instanceNum = Integer.parseInt(getValue().getStringValue());
+            if (instanceNum > 96) {
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SetVar.java:160]: change 96 be a fe config

**Comment by Mryange** [comment]
run p0

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**